src/simplify_algebra.cpp

```diff
@@ -1446,10 +1446,13 @@ struct find_split_transpose
         {
             return;
         }
+        if(std::any_of(split_outputs.begin(), split_outputs.end(), [](auto i) {
+               return i->outputs().size() != 1;
+           }))
+            return;
```

Comment:
I think can add `match::used_once()` to the matcher instead.

Reply:
I'll have to do it for all the slices, I dont' think i can do it inside matcher.

Reply:
I think you could use `match::any_of[match::outputs()]` but that might be more complicated.